src/environment.cc

```diff
@@ -179,6 +180,29 @@ const std::string& Environment::KubernetesClusterName() const {
   return kubernetes_cluster_name_;
 }
 
+const std::string& Environment::KubernetesClusterLocation() const {
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (kubernetes_cluster_location_.empty()) {
+    if (!config_.KubernetesClusterLocation().empty()) {
+      kubernetes_cluster_location_ = config_.KubernetesClusterLocation();
+    } else {
+      // Get the kube-env.
+      const std::string kube_env =
+          GetMetadataString("instance/attributes/kube-env");
```

Comment:
For all clusters with Kubernetes in version 1.10.0 or newer, Metadata Server exports instance attribute "cluster-location". This works also for all GKE clusters created or upgraded after last GKE release (with cl/185806663) included. You shouldn't need to extract this from "kube-env". Do we need to support older clusters?

I think we discussed this earlier in email thread, sorry if I didn't clarify this enough.

Reply:
Sorry, I didn't find the email thread in time. Updated the bug with the info and fixed.